docs/concepts/plans.md

```diff
@@ -68,29 +72,69 @@ When a plan is applied to an environment, the environment gets associated with t
+Despite all the benefits, the approach described above is not without trade-offs.
 
-At the moment, we are using the term backfilling broadly to describe any situation in which a model is updated. That includes these operations:
+When a new model version is just created, a physical table assigned to it is empty. Therefore, SQLMesh needs to re-apply the logic of the new model version to the entire date range of this model in order to populate the new version's physical table. This process is called backfilling.
+
+We use the term backfilling broadly to describe any situation in which a model is updated. That includes these operations:
 
 * When a VIEW model is created
 * When a FULL model is built
 * When an INCREMENTAL model is built for the first time
 * When an INCREMENTAL model has recent data appended to it
 * When an INCREMENTAL model has older data inserted (i.e., resolving a data gap or prepending historical data)
 
-We will be iterating on terminology to better capture the nuances of each type in future versions.
+Note for incremental models: despite the fact that backfilling can happen incrementally (see `batch_size` parameter on models), there is an extra cost associated with this operation due to additional runtime involved. If the runtime cost is a concern, use a [forward-only plan](#forward-only-plans) instead.
+
+### Virtual Update
+A benefit of SQLMesh's approach is that data for a new model version can be fully pre-built while still in a development environment. That way all changes and their downstream dependencies can be fully previewed before they are promoted to the production environment.
+
+With this approach, the process of promoting a change to production is reduced to reference swapping.
+
+If during plan creation no data gaps have been detected and only references to new model versions need to be updated, then the update is referred to as a Virtual Update. Virtual Updates impose no additional runtime overhead or cost.
+
+### Start and end dates
+
+The `plan` command provides two temporal options: `--start` and `--end`. These options are only applicable to plans for non-prod environments.
+
+For context, every model has a start date. The start can be specified in [the model definition](./models/overview.md#start), in the [project configuration's `model_defaults`](../guides/configuration.md#model-defaults), or by SQLMesh's default value of yesterday.
+
+Because the prod environment supports business operations, prod plans ensure every model is backfilled from its start date until the most recent completed time interval. The `plan` command's `--start` and `--end` options are not applicable for prod due to that restriction.
```

Comment:
> The `plan` command's `--start` and `--end` options are not applicable for prod due to that restriction.

This is not entirely accurate. For example, you can specify start and end if you're restating in prod